.github/workflows/publish.yml

```
@@ -0,0 +1,35 @@
+name: Build Jekyll site
+
+on:
+  push:
+    branches: ["main"]
```

As you mentioned on the issue
```suggestion
    branches: ["gh-pages"]
```